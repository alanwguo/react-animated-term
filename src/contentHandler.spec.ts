// Copyright 2022, Nitric Technologies Pty Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
import termContent from './contentHandler';
import { Line } from './Renderer';

test('correctly handles empty content', () => {
  const lines = [];
  const content = termContent(lines);
  const { value, done } = content.next();
  expect(value).toEqual([]);
  expect(done).toBe(true);
});

test('correctly handles static cmd line', () => {
  const lines = [
    {
      text: 'ls',
      cmd: true,
    },
  ];
  const content = termContent(lines);
  let next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(false);

  next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(false);

  next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(false);

  next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(false);

  next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(true);
});

test('correctly handles static output line', () => {
  const lines = [
    {
      text: 'foo',
      cmd: false,
    },
  ];
  const content = termContent(lines);
  let next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(false);

  next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(false);

  next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(true);
});

test('correctly handles framed output', () => {
  const lines: Line[] = [
    {
      cmd: false,
      frames: [
        {
          text: '[------------------------------------------------] 0/100',
        },
        {
          text: '[################################################] 100/100',
        },
      ],
    },
  ];

  const content = termContent(lines);
  let next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(false);

  next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(false);

  next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(false);

  next = content.next();
  expect(next.value).toMatchSnapshot();
  expect(next.done).toBe(true);
});
