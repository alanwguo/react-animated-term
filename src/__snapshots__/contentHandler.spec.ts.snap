/*
Copyright 2022, Nitric Technologies Pty Ltd.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly handles framed output 1`] = `
Array [
  Object {
    "cmd": false,
    "current": true,
    "id": 0,
    "text": "[------------------------------------------------] 0/100",
  },
]
`;

exports[`correctly handles framed output 2`] = `
Array [
  Object {
    "cmd": false,
    "current": true,
    "id": 0,
    "text": "[################################################] 100/100",
  },
]
`;

exports[`correctly handles framed output 3`] = `
Array [
  Object {
    "cmd": false,
    "current": false,
    "id": 0,
    "text": "[################################################] 100/100",
  },
]
`;

exports[`correctly handles framed output 4`] = `
Array [
  Object {
    "cmd": false,
    "current": false,
    "id": 0,
    "text": "[################################################] 100/100",
  },
]
`;

exports[`correctly handles static cmd line 1`] = `
Array [
  Object {
    "cmd": true,
    "current": true,
    "id": 0,
    "text": "",
  },
]
`;

exports[`correctly handles static cmd line 2`] = `
Array [
  Object {
    "cmd": true,
    "current": true,
    "id": 0,
    "text": "l",
  },
]
`;

exports[`correctly handles static cmd line 3`] = `
Array [
  Object {
    "cmd": true,
    "current": true,
    "id": 0,
    "text": "ls",
  },
]
`;

exports[`correctly handles static cmd line 4`] = `
Array [
  Object {
    "cmd": true,
    "current": true,
    "id": 0,
    "text": "ls",
  },
]
`;

exports[`correctly handles static cmd line 5`] = `
Array [
  Object {
    "cmd": true,
    "current": true,
    "id": 0,
    "text": "ls",
  },
]
`;

exports[`correctly handles static output line 1`] = `
Array [
  Object {
    "cmd": false,
    "current": false,
    "id": 0,
    "text": "foo",
  },
]
`;

exports[`correctly handles static output line 2`] = `
Array [
  Object {
    "cmd": false,
    "current": false,
    "id": 0,
    "text": "foo",
  },
]
`;

exports[`correctly handles static output line 3`] = `
Array [
  Object {
    "cmd": false,
    "current": false,
    "id": 0,
    "text": "foo",
  },
]
`;
